src/Ar/Chopper/ChopRender.cpp

```diff
@@ -228,3 +231,46 @@ signed long appendTo(UDINT pdest, UDINT dsize, UDINT* poffset, UDINT psource, UD
 	
 	return CHOP_ERR_DEST_LENGTH;
 }
+
+
+/** postProcessRealTypeStr
+* Checks whether real-string starts with "." or "-." and adds a "0"
+* E.g. ".123" and "-.456" are converted to "0.123" and "-0.456", respectively
+*/
+signed long postProcessRealTypeStr(plcstring* buffer, unsigned long bufferSize) {
+	
+	if ((buffer == 0) || bufferSize < 2) {
+		// Error: Invalid inputs
+		return -1;
+	}
+	
+	int operation = 0;
+	if (buffer[0] == '.') 							operation = 1;	// begins with "."
+	if ((buffer[0] == '-') && (buffer[1] == '.'))	operation = 2;	// begins with "-."
+	
+	if (operation > 0) {
+		signed long lenWithNullTerm = strlen(buffer) + 1;	// Note: strlen does NOT include null termination character
```

Comment:
StringExt may have a safer strlen (in the event that buffer is not null terminated).
If so, I think we should use it.

Reply:
Changed to `brsstrlen`, the only alternative I could find. Not sure if that's safer though, or just different.

Reply:
I had some kind of max-length checking function in mind.
strnlen

I'm not sure we have access to something like this